packages/arkor-local/python/mlx/dataset_prep.py

```
@@ -0,0 +1,254 @@
+            raise DatasetPrepError(
+                f"unsupported sharegpt speaker {speaker!r} (expected one of "
+                f"{sorted(_SHAREGPT_ROLES)})"
+            )
+        content = turn.get("value") if "value" in turn else turn.get("content")
+        messages.append({"role": role, "content": content})
+    return messages
+
+
+def _normalise_messages(messages) -> list:
+    if not isinstance(messages, list) or not messages:
+        raise DatasetPrepError("chatml rows must carry a non-empty message list")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)